sql/parser/builtins.go

```diff
@@ -963,15 +980,20 @@ func datumToString(datum Datum) (string, error) {
 
 var replaceSubRe = regexp.MustCompile(`\\[&1-9]`)
 
-func regexpReplace(s, pattern, to, sqlFlags string) (Datum, error) {
+func regexpReplace(ctx EvalContext, s, pattern, to, sqlFlags string) (Datum, error) {
 	pattern, global, err := regexpEvalFlags(pattern, sqlFlags)
 	if err != nil {
 		return nil, err
 	}
 
-	patternRe, err := regexp.Compile(pattern)
-	if err != nil {
-		return nil, err
+	var patternRe *regexp.Regexp
+	if patternRe = ctx.ReCache.Lookup(pattern); patternRe == nil {
```

Comment:
Since `ReCache` is already specific to the `Regexp` type, we could encapsulate the lookup/compile/update pattern in a method `ReCache.GetRegexp(string)`. This would also guard against any callers that create regexps in a way that differs in ways not captured by the cache key (i.e. using `CompilePOSIX` instead of `Compile`).